SpatialGDK/Source/SpatialGDKServices/Public/LocalReceptionistProxyServerManager.h

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "CoreMinimal.h"
+
+DECLARE_LOG_CATEGORY_EXTERN(LogLocalReceptionistProxyServerManager, Log, All);
+
+
+class FLocalReceptionistProxyServerManager
+{
+public:
+	FLocalReceptionistProxyServerManager();
+
+	bool KillProcessBlockingPort();
+
+	bool SPATIALGDKSERVICES_API TryStartReceptionistProxyServer(bool bIsRunningInChina, const FString& CloudDeploymentName);
+	bool SPATIALGDKSERVICES_API TryStopReceptionistProxyServer();
+
+private:
+	FProcHandle ProxyServerProcHandle;
+	FString RunningCloudDeploymentName;
+
+	static const int32 ExitCodeSuccess = 0;
+	static const int32 RequiredRuntimePort = 7777;
```

Comment:
will that port ever change? (I'm not sure myself, so asking this here)

Reply:
At the moment I run the proxy with the default parameters that are` --listening_address "127.0.0.1"` and `--local_receptionist_port 7777` . Is there a case where we will need the user to select the port?